src/util/check.cpp

```diff
@@ -27,3 +28,9 @@ void assertion_fail(const char* file, int line, const char* func, const char* as
     fwrite(str.data(), 1, str.size(), stderr);
     std::abort();
 }
+
+void unreachable_fail(const char* file, int line, const char* func)
+{
+    std::cerr << strprintf("Internal bug detected: Unreachable code reached (fatal, aborting)\n%s:%d (%s)\nPlease report this issue here: %s\n", file, line, func, PACKAGE_BUGREPORT) << std::endl;
```

Comment:
rebase and use the existing string template?

Reply:
[Done](https://github.com/bitcoin/bitcoin/pull/26504#issuecomment-1345356487).